src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.Prerendering.cs

```diff
@@ -33,19 +35,45 @@ protected override IComponent ResolveComponentForRenderMode([DynamicallyAccessed
         }
     }
 
+    public PersistedStateSerializationMode GetComponentSerializationMode(IComponent component)
+    {
+        var ssrRenderBoundary = GetClosestRenderModeBoundary(component);
+        var renderMode = ssrRenderBoundary!.RenderMode;
```

Comment:
What happens if there's no render mode boundary? This can happen if a render mode wasn't set in an ancestor component.

Reply:
Thanks for noting! I wanted to get back to it and throw an error and forgot. Fixed now